Duplicate addition of cocotb_iverilog_dump.v in VERILOG_SOURCES when using make -f

make version = 3.81

When using `make -f` to invoke the Makefile.sim, the file `cocotb_iverilog_dump.v` 
is appended multiple times to the `VERILOG_SOURCES` variable. This causes duplicate module 
declarations and results in compilation errors during simulation.

**Reproduce**
I modify [quckstart](https://github.com/cocotb/cocotb/tree/master/examples/doc_examples/quickstart) Makefile as follows:

```makefile
# This file is public domain, it can be freely copied without restrictions.
# SPDX-License-Identifier: CC0-1.0

# Makefile

# defaults
SIM ?= icarus
TOPLEVEL_LANG ?= verilog
WAVES = 1

VERILOG_SOURCES += $(PWD)/my_design.sv
# use VHDL_SOURCES for VHDL files

# TOPLEVEL is the name of the toplevel module in your Verilog or VHDL file
TOPLEVEL = my_design
WAVES = 1

# MODULE is the basename of the Python test file
MODULE = test_my_design

COCOTB_MAKEFILE = $(shell poetry run cocotb-config --makefiles)/Makefile.sim
export
# include cocotb's make rules to take care of the simulator setup
sim:
	poetry run \
	make \
	-f $(COCOTB_MAKEFILE)

```

and run `make sim` and get
```
poetry run \
	make \
	-f /Users/yizishun/Library/Caches/pypoetry/virtualenvs/cocotb-try-fuIVHJhZ-py3.12/lib/python3.12/site-packages/cocotb/share/makefiles/Makefile.sim
rm -f results.xml
"/Library/Developer/CommandLineTools/usr/bin/make" -f /Users/yizishun/Library/Caches/pypoetry/virtualenvs/cocotb-try-fuIVHJhZ-py3.12/lib/python3.12/site-packages/cocotb/share/makefiles/Makefile.sim results.xml
mkdir -p sim_build
/etc/profiles/per-user/yizishun/bin/iverilog -o sim_build/sim.vvp -D COCOTB_SIM=1 -s my_design -g2012 -f sim_build/cmds.f -s cocotb_iverilog_dump  /Users/yizishun/test/cocotb_try/my_design.sv sim_build/cocotb_iverilog_dump.v sim_build/cocotb_iverilog_dump.v
sim_build/cocotb_iverilog_dump.v:1: error: 'cocotb_iverilog_dump' has already been declared in this scope.
sim_build/cocotb_iverilog_dump.v:1:      : It was declared here as a module.
sim_build/cocotb_iverilog_dump.v:6: error: Module cocotb_iverilog_dump was already declared here: sim_build/cocotb_iverilog_dump.v:1

make[2]: *** [sim_build/sim.vvp] Error 2
make[1]: *** [sim] Error 2
make: *** [sim] Error 2
```

--- a/docs/source/newsfragments/5382.change.rst
+++ b/docs/source/newsfragments/5382.change.rst
@@ -1,0 +1,1 @@
+The :ref:`Makefile <building>` target ``sim`` no longer recursively calls ``make``. This makes it possible to use ``+=`` with Make variables without the right hand side being added twice.

--- a/src/cocotb_tools/makefiles/Makefile.inc
+++ b/src/cocotb_tools/makefiles/Makefile.inc
@@ -11,7 +11,7 @@ COCOTB_MAKEFILE_INC_INCLUDED = 1
 
 # Default sim rule will force a re-run of the simulation (though the cocotb library
 # and RTL compilation phases are still evaluated by makefile dependencies)
-.PHONY: sim, _rm_regresion
+.PHONY: sim, _rm_results
 _rm_results:
 	$(RM) $(COCOTB_RESULTS_FILE)
 sim: _rm_results regression

--- a/src/cocotb_tools/makefiles/Makefile.inc
+++ b/src/cocotb_tools/makefiles/Makefile.inc
@@ -11,10 +11,10 @@ COCOTB_MAKEFILE_INC_INCLUDED = 1
 
 # Default sim rule will force a re-run of the simulation (though the cocotb library
 # and RTL compilation phases are still evaluated by makefile dependencies)
-.PHONY: sim
-sim:
+.PHONY: sim, _rm_regresion
+_rm_results:
 	$(RM) $(COCOTB_RESULTS_FILE)
-	"$(MAKE)" -f $(firstword $(MAKEFILE_LIST)) $(COCOTB_RESULTS_FILE)
+sim: _rm_results regression
 
 # Make sure to use bash for the pipefail option used in many simulator Makefiles
 SHELL := bash

